src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -463,6 +461,7 @@ private void SetTransferMode(TransferMode transferMode)
                 await _receiveLoopTask;
             }
 
+            _closedTcs.TrySetResult(null);
```

Comment:
help me understand why we have this new code?

Reply:
Since closed is now a task, we're making sure to complete it in dispose